tools/rules-testing/tests/nginx.ini

```diff
@@ -1,79 +1,74 @@
 ; YYYY/MM/DD HH:MM:SS [LEVEL] PID:TID yadda yadda
 [Nginx messages grouped.]
 log 1 pass = 2014/12/30 06:07:37 [yadda] 80:2 yadda yadda
-
 rule = 31300
 alert = 0
 decoder = nginx-errorlog
 
 [Nginx error message.]
 log 1 pass = 2014/12/30 06:07:37 [error] 80:2 yadda yadda
-
 rule = 31301
 alert = 3
 decoder = nginx-errorlog
 
 [Nginx warning message.]
 log 1 pass = 2014/12/30 06:07:37 [warn] 80:2 yadda yadda
-
 rule = 31302
 alert = 3
 decoder = nginx-errorlog
 
 [Nginx critical message.]
 log 1 pass = 2014/12/30 06:07:37 [crit] 80:2
-
 rule = 31303
 alert = 5
 decoder = nginx-errorlog
 
 [Server returned 404 (reported in the access.log).]
 log 1 pass = 2015/01/08 11:31:23 [error] 80:2 blah blah failed (2: No such file or directory)
 log 2 pass = 2015/01/08 11:31:23 [error] 80:2 blah blah is not found (2: No such file or directory)
-
 rule = 31310
 alert = 0
 decoder = nginx-errorlog
 
 [Incomplete client request.]
 log 1 pass = 2015/01/08 11:31:23 [error] 80:2 blah blah accept() failed (53: Software caused connection abort)
-
 rule = 31311
 alert = 0
 decoder = nginx-errorlog
 
 [Initial 401 authentication request.]
 log 1 pass = 2015/01/08 11:31:23 [error] 80:2 no user/password was provided for basic authentication
-
 rule = 31312
 alert = 0
 decoder = nginx-errorlog
 
 [Web authentication failed.]
 log 1 pass = 2015/01/08 11:31:23 [error] 80:2 yadda password mismatch, client yadda
 log 2 pass = 2015/01/08 11:31:23 [error] 80:2 yadda was not found in yadda
-
 rule = 31315
 alert = 5
 decoder = nginx-errorlog
 
 # Can't yet test frequency   <rule id="31316" level="10" frequency="6" timeframe="240">
 ;[Multiple web authentication failures.]
-;
 ;rule = 31316
 ;alert = 10
 ;decoder = nginx-errorlog
 
 [Common cache error when files were removed.]
 log 1 pass = 2015/01/08 11:31:23 [crit] 80:2 yadda yadda failed (2: No such file or directory
-
 rule = 31317
 alert = 0
 decoder = nginx-errorlog
 
 [Invalid URI, file name too long.]
 log 1 pass = 2015/01/08 11:31:23 [error] 80:2 yadda yadda failed (36: File name too long)
-
 rule = 31320
 alert = 10
 decoder = nginx-errorlog
+
+[NAXSI warning]
+log 1 pass = 2013/11/10 07:36:19 [error] 8278#0: *5932 NAXSI_FMT: ip=X.X.X.X&server=Y.Y.Y.Y&uri=/phpMyAdmin-2.8.2/scripts/setup.php&learning=0&vers=0.52&total_processed=472&total_blocked=204&block=0&cscore0=$UWA&score0=8&zone0=HEADERS&id0=42000227&var_name0=user-agent, client: X.X.X.X, server: blog.memze.ro, request: "GET /phpMyAdmin-2.8.2/scripts/setup.php HTTP/1.1", host: "X.X.X.X"
```

Comment:
This check works fine:

```
# ossec-logtest
2019/08/23 04:11:42 ossec-testrule: INFO: Started (pid: 40767).
ossec-testrule: Type one log per line.

2013/11/10 07:36:19 [error] 8278#0: *5932 NAXSI_FMT: ip=X.X.X.X&server=Y.Y.Y.Y&uri=/phpMyAdmin-2.8.2/scripts/setup.php&learning=0&vers=0.52&total_processed=472&total_blocked=204&block=0&cscore0=$UWA&score0=8&zone0=HEADERS&id0=42000227&var_name0=user-agent, client: X.X.X.X, server: blog.memze.ro, request: "GET /phpMyAdmin-2.8.2/scripts/setup.php HTTP/1.1", host: "X.X.X.X"


**Phase 1: Completed pre-decoding.
       full event: '2013/11/10 07:36:19 [error] 8278#0: *5932 NAXSI_FMT: ip=X.X.X.X&server=Y.Y.Y.Y&uri=/phpMyAdmin-2.8.2/scripts/setup.php&learning=0&vers=0.52&total_processed=472&total_blocked=204&block=0&cscore0=$UWA&score0=8&zone0=HEADERS&id0=42000227&var_name0=user-agent, client: X.X.X.X, server: blog.memze.ro, request: "GET /phpMyAdmin-2.8.2/scripts/setup.php HTTP/1.1", host: "X.X.X.X"'
       timestamp: '(null)'
       hostname: 'ubuntu'
       program_name: '(null)'
       log: '2013/11/10 07:36:19 [error] 8278#0: *5932 NAXSI_FMT: ip=X.X.X.X&server=Y.Y.Y.Y&uri=/phpMyAdmin-2.8.2/scripts/setup.php&learning=0&vers=0.52&total_processed=472&total_blocked=204&block=0&cscore0=$UWA&score0=8&zone0=HEADERS&id0=42000227&var_name0=user-agent, client: X.X.X.X, server: blog.memze.ro, request: "GET /phpMyAdmin-2.8.2/scripts/setup.php HTTP/1.1", host: "X.X.X.X"'

**Phase 2: Completed decoding.
       decoder: 'nginx-errorlog'
       srcip: 'X.X.X.X'
       server: 'Y.Y.Y.Y'
       uri: '/phpMyAdmin-2.8.2/scripts/setup.php'
       learning: '0'
       vers: '0.52'
       total_processed: '472'
       total_blocked: '204'
       block: '0'
       attack: '$UWA'
       score: '8'

**Phase 3: Completed filtering (rules).
       Rule id: '31334'
       Level: '3'
       Description: 'NAXSI warning'
**Alert to be generated.
```

However, the `runtests.py` script fails when parsing it:

```
- [ File = ./tests/nginx.ini ] ---------
Traceback (most recent call last):
  File "runtests.py", line 104, in <module>
    OT.run(selective_test)
  File "runtests.py", line 72, in run
    tGroup.read([aFile])
  File "/usr/lib/python2.7/ConfigParser.py", line 305, in read
    self._read(fp, filename)
  File "/usr/lib/python2.7/ConfigParser.py", line 546, in _read
    raise e
ConfigParser.ParsingError: File contains parsing errors: ./tests/nginx.ini
	[line 82]: ' log 1 pass = 2013/11/10 07:36:19 [error] 8278#0: *5932 NAXSI_FMT: ip=X.X.X.X&server=Y.Y.Y.Y&uri=/phpMyAdmin-2.8.2/scripts/setup.php&learning=0&vers=0.52&total_processed=472&total_blocked=204&block=0&cscore0=$UWA&score0=8&zone0=HEADERS&id0=42000227&var_name0=user-agent, client: X.X.X.X, server: blog.memze.ro, request: "GET /phpMyAdmin-2.8.2/scripts/setup.php HTTP/1.1", host: "X.X.X.X"\n'
	[line 83]: ' rule = 31334\n'
	[line 84]: ' alert = 3\n'
	[line 85]: ' decoder = nginx-errorlog\n'
```

Reply:
Forget it. My fault copying the tests.